src/service.rs

```diff
@@ -98,6 +101,38 @@ where
     pub fn take_snapshot(&mut self) {
         self.frugalos_segment_service.take_snapshot();
     }
+
+    /// Inspects a physical device.
+    /// `PhysicalDeviceInspection` contains a sequence of `LumpId`s.
+    pub fn inspect_physical_device(
+        &self,
+        device_id: entity::device::DeviceId,
+    ) -> impl Future<Item = PhysicalDeviceInspection, Error = Error> {
+        let future = self
+            .local_devices
+            .values()
+            .find_map(|d| {
+                if d.id().as_str() == device_id {
+                    d.handle()
+                } else {
+                    None
+                }
+            }).ok_or_else(|| ErrorKind::InvalidInput.into())
+            .into_future();
+
+        future.and_then(|handle| {
+            handle
+                .request()
+                .list()
```

Comment:
[For the sake of understanding]
Here we issue the `list` command to a device that has the given device-id passed as the command-line argument.
This line is one of the core parts of the present PR; right?

Reply:
Yes.